### ISSUE TITLE
questions about loss

### ISSUE THREAD

**Issue by annopackage** [OpenGVLab/InternVL#769]
Hi, thanks for your great open-sourcing work.
Regarding the tensorboard here https://huggingface.co/OpenGVLab/InternVL2_5-8B/tensorboard, I wonder why the loss increases at around 0.7 epoch. Can you share some details about the performance between the intermediate step with the minimal loss and final checkpoint?

**Comment by czczup** [comment]
The loss initially decreasing and then increasing might be due to the dynamic changes in data distribution introduced by the data packing strategy during training. 

At the early stage, the model learns from a relatively uniform data distribution, leading to a decrease in loss. 

However, as training progresses and the dataset nears depletion, the Dataloader begins to empty its buffer, causing a shift in the data distribution received by the model in the later stages of training, which results in changes in the loss trend. This might indicate that some edge cases have not been properly handled in our code.

Nevertheless, we have not observed significant drawbacks of this phenomenon. For instance, the lowest loss occurs at approximately 4000 iterations, while the model is ultimately trained up to 5300–5400 iterations. In our evaluation, the final weights outperform the intermediate weights at 4000 iterations.
